u-root.go

```diff
@@ -287,20 +290,36 @@ func Main(l ulog.Logger, buildOpts *gbbgolang.BuildOpts) error {
 
 		// Resolve globs into package imports.
 		//
-		// Currently allowed formats:
-		//   Go package imports; e.g. github.com/u-root/u-root/cmds/ls (must be in $GOPATH)
+		// Currently allowed format:
 		//   Paths to Go package directories; e.g. $GOPATH/src/github.com/u-root/u-root/cmds/*
+		//   u-root templates; e.g. all, core, minimal (requires uroot-source)
+		//   Import paths of u-root commands; e.g. github.com/u-root/u-root/cmds/* (requires uroot-source)
 		var pkgs []string
 		for _, a := range flag.Args() {
 			p, ok := templates[a]
 			if !ok {
+				if !validateArg(a) {
+					l.Printf("%q is not a valid path", a)
+					continue
+				}
 				pkgs = append(pkgs, a)
 				continue
 			}
+			if *urootSourceDir != "" {
+				for _, pkg := range p {
+					pkgs = append(pkgs, filepath.Join(*urootSourceDir, pkg))
```

Comment:
this doesn't seem necessarily correct -- what if I want to do the following?

u-root -uroot-source=$HOME/u-root cmds/core/ip ./foobar/cmd/u-bmc

isn't this going to apply the source dir to every listed package?

Reply:
IIRC it should try both variation, checking for the code directly in the provided path and with the uroot-source prefixed. I'll check the code again to make sure though.